### PR TITLE
5.3.1

### PR DESCRIPTION
**Pull request by rndmh3ro**
## Problem
- App does not install correctly

## Solution
- Fix various scripts
- superseeds #5 

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Package_check results
---
*If you have access to [App Continuous Integration for packagers](https://yunohost.org/#/packaging_apps_ci) you can provide a link to the package_check results like below, replacing '-NUM-' in this link by the PR number and USERNAME by your username on the ci-apps-dev. Or you provide a screenshot or a pastebin of the results*

[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/invoiceninja_ynh%20PR-NUM-%20(USERNAME)/)

### CODE REVIEW

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

File: check_process [hunk 1]
```diff
@@ -1,30 +1,23 @@
...
+		domain="domain.tld"
+		path="/path"
+		admin="john"
 		language="fr"
+                is_public=1
+                password="1Strong-Password"
 	; Checks
 		pkg_linter=1
-		setup_sub_dir=1
+		setup_sub_dir=0
```
Comment (alexAubin) [inline check_process:11]: Don't we want to enable this test ? (c.f. your post on the forum)

File: conf/default.env [hunk 2]
```diff
@@ -0,0 +1,58 @@
+APP_NAME="Invoice Ninja"
+APP_ENV=production
+APP_KEY=base64:3E5HVZ/T+VbZu+oA4GNlwnvwFe3jNtkGbR95K0uwr7A=
```
Comment (alexAubin) [inline conf/default.env:3]: This looks like a secret somehow ? Isn't it necessary to generate it during install ?

File: conf/default.env [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#options - snappdf / phantom / hosted_ninja
+PDF_GENERATOR=phantom 
+
+PHANTOMJS_KEY='__PHANTOMJS_KEY__'
+PHANTOMJS_SECRET=__APP_KEY__
+
+UPDATE_SECRET=__API_SECRET__
+
+COMPOSER_AUTH='{"github-oauth": {"github.com": "${{ secrets.GITHUB_TOKEN }}"}}'
+SENTRY_LARAVEL_DSN=https://cc7e8e2c678041689e53e409b7dba236@sentry.invoicing.co/5
```
Comment (alexAubin) [inline conf/default.env:56]: This looks weird too ... is it okay to have this token here ?

File: manifest.json [hunk 4]
```diff
@@ -3,54 +3,47 @@
...
     "url": "https://invoiceninja.org",
+    "upstream": {
+        "license": "ELv2",
+        "website": "https://invoiceninja.org",
+        "demo": "https://app.invoiceninja.com/dashboard",
+        "admindoc": "https://invoiceninja.github.io/",
+        "userdoc": "https://invoiceninja.github.io/",
+        "code": "https://github.com/invoiceninja/invoiceninja"
+    },
     "license": "AAL",
```
Comment (alexAubin) [inline manifest.json:18]: Is license AAL, or ELv2 ?

File: scripts/upgrade [hunk 5]
```diff
@@ -56,135 +48,117 @@ ynh_script_progression --message="Backing up the app before upgrading (may take
...
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
 
-config="$final_path/.env"
-ynh_backup_if_checksum_is_different --file="$final_path/CONFIG_FILE"
-
 if [ "$upgrade_type" == "UPGRADE_APP" ]
 then
-	ynh_script_progression --message="Upgrading source files..."
+	ynh_script_progression --message="Upgrading source files..." --time --weight=1
```
Comment (alexAubin) [inline scripts/upgrade:71]:
```suggestion
	ynh_script_progression --message="Upgrading source files..."
```

File: check_process (hunk 1 above)
Comment (rndmh3ro) [reply check_process:11]: Sure, if it's working. :) Right now I can't get it to work.

File: conf/default.env (hunk 2 above)
Comment (rndmh3ro) [reply conf/default.env:3]: Yeah, you're right. I'll fix it.

File: conf/default.env (hunk 3 above)
Comment (rndmh3ro) [reply conf/default.env:56]: You're right, I copy-pasted this from somewhere. Will remove it.

File: manifest.json (hunk 4 above)
Comment (rndmh3ro) [reply manifest.json:18]: It's ELv2. It was changed recently.

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by rndmh3ro** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by rndmh3ro** [comment]
@ericgaspar, the PR is hanging on this problem: https://forum.yunohost.org/t/help-getting-subdir-installation-to-work/17368/2

**Comment by rndmh3ro** [comment]
superseeded by #7

**Pull request closed without merging**